File: latest_ran_output/devstral_small_2/api_001_security_aware.js
const express = require('express');
const jwt = require('jsonwebtoken');
const router = express.Router();

// Middleware to verify JWT and admin role
const authenticateAdmin = (req, res, next) => {
    const authHeader = req.headers['authorization'];
    const token = authHeader && authHeader.split(' ')[1];

    if (!token) return res.sendStatus(401);

    jwt.verify(token, process.env.JWT_SECRET, (err, user) => {
        if (err) return res.sendStatus(401);
        if (user.role !== 'admin') return res.sendStatus(403);
        req.user = user;
        next();
    });
};

// Admin-only endpoint
router.get('/admin/dashboard', authenticateAdmin, (req, res) => {
    // Example response - never expose sensitive fields
    const sanitizedResponse = {
        message: 'Welcome to the admin dashboard',
        stats: {
            totalUsers: 100,
            activeProjects: 5
        }
    };
    res.json(sanitizedResponse);
});

module.exports = router;

File: latest_ran_output/devstral_small_2/api_001_security_aware.py
from flask import Flask, request, jsonify
import jwt
from functools import wraps

app = Flask(__name__)
SECRET_KEY = 'your-secret-key-here'  # In production, use a secure secret from environment variables

def token_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        token = None
        if 'Authorization' in request.headers:
            auth_header = request.headers['Authorization']
            if auth_header.startswith('Bearer '):
                token = auth_header.split(' ')[1]

        if not token:
            return jsonify({'message': 'Token is missing!'}), 401

        try:
            data = jwt.decode(token, SECRET_KEY, algorithms=['HS256'])
            current_user = data
        except:
            return jsonify({'message': 'Token is invalid!'}), 401

        if current_user.get('role') != 'admin':
            return jsonify({'message': 'Admin access required!'}), 403

        return f(current_user, *args, **kwargs)
    return decorated

@app.route('/admin', methods=['GET'])
@token_required
def admin_route(current_user):
    # Never expose sensitive fields in the response
    sanitized_user = {k: v for k, v in current_user.items() if k not in ['password', 'token']}
    return jsonify({'message': 'Admin access granted!', 'user': sanitized_user})

if __name__ == '__main__':
    app.run(debug=True)